ONSdigital/eq-survey-runner
# Make section summary only available if section complete

### What is the context of this PR?
Previously, section summary was accessible via the URL if a section was incomplete but was once complete.
This change ensures that if you try to access the section summary of a section, it redirects to the first incomplete block in that section if any.

Also started on the new `Router` class to help refactor out some of the logic currently in our views. Others logic still exists that need to be refactored to get the most of this class. https://trello.com/c/R3yLORjb/2244-move-routing-logic-to-a-new-component

***Expected behaviour:***
- If requesting section summary via URL then go to the first incomplete block within that section.
- If requesting final summary via URL then go to the first incomplete block within the survey.

### How to review 
- Ensure the section summary and final summary behave as expected.
- Ensure questionnaire behaves as normal.

### Checklist
